src/pointer.rs

```diff
@@ -0,0 +1,427 @@
+            contact_geometry: Size::new(1., 1.),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum PointerType {
+    Mouse(MouseInfo),
+    Pen(PenInfo),
+    Eraser(PenInfo),
+    Touch(TouchInfo),
+    // Apple has force touch devices that provide pressure info, but nothing further.
+    // Assume that that may become more of a thing in the future?
+}
+
+/// An indicator of which pointer button was pressed.
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+#[repr(u8)]
+pub enum PointerButton {
+    /// No mouse button.
+    // MUST BE FIRST (== 0)
+    None,
+    /// Left mouse button, Left Mouse, Touch Contact, Pen contact.
+    Left,
+    /// Right mouse button, Right Mouse, Pen barrel button.
+    Right,
+    /// Middle mouse button.
+    Middle,
+    /// X1 (back) Mouse.
+    X1,
+    /// X2 (forward) Mouse.
+    X2,
+}
+
+impl From<crate::MouseButton> for PointerButton {
+    fn from(m: crate::MouseButton) -> Self {
+        match m {
+            crate::MouseButton::None => PointerButton::None,
+            crate::MouseButton::Left => PointerButton::Left,
+            crate::MouseButton::Right => PointerButton::Right,
+            crate::MouseButton::Middle => PointerButton::Middle,
+            crate::MouseButton::X1 => PointerButton::X1,
+            crate::MouseButton::X2 => PointerButton::X2,
+        }
+    }
+}
+
+impl PointerButton {
+    /// Returns `true` if this is [`PointerButton::Left`].
+    ///
+    /// [`MouseButton::Left`]: #variant.Left
+    #[inline]
+    pub fn is_left(self) -> bool {
+        self == PointerButton::Left
+    }
+
+    /// Returns `true` if this is [`PointerButton::Right`].
+    ///
+    /// [`PointerButton::Right`]: #variant.Right
+    #[inline]
+    pub fn is_right(self) -> bool {
+        self == PointerButton::Right
+    }
+
+    /// Returns `true` if this is [`PointerButton::Middle`].
+    ///
+    /// [`PointerButton::Middle`]: #variant.Middle
+    #[inline]
+    pub fn is_middle(self) -> bool {
+        self == PointerButton::Middle
+    }
+
+    /// Returns `true` if this is [`PointerButton::X1`].
+    ///
+    /// [`PointerButton::X1`]: #variant.X1
+    #[inline]
+    pub fn is_x1(self) -> bool {
+        self == PointerButton::X1
+    }
+
+    /// Returns `true` if this is [`PointerButton::X2`].
+    ///
+    /// [`PointerButton::X2`]: #variant.X2
+    #[inline]
+    pub fn is_x2(self) -> bool {
+        self == PointerButton::X2
+    }
+}
+
+/// A set of [`PointerButton`]s.
+///
+/// [`PointerButton`]: enum.PointerButton.html
+#[derive(PartialEq, Eq, Clone, Copy, Default)]
+pub struct PointerButtons(u8);
+
+impl PointerButtons {
+    /// Create a new empty set.
+    #[inline]
+    pub fn new() -> PointerButtons {
+        PointerButtons(0)
+    }
+
+    /// Add the `button` to the set.
+    #[inline]
+    pub fn insert(&mut self, button: PointerButton) {
+        self.0 |= 1.min(button as u8) << button as u8;
```

Comment:
we write `1.min(button as u8) << button as u8` a lot, maybe it could be a method on button? since there's only 6 cases, could also consider writing out the 6 possible values? (although i suppose this maybe shows why they have these values a little more clearly?)

Reply:
I wrote out the match, and I'm not really sure which I prefer :shrug:  Anyway, it's certainly worth having it in a function